src/nanoarrow/array.c

```diff
@@ -995,6 +1021,31 @@ static int ArrowArrayViewValidateDefault(struct ArrowArrayView* array_view,
         }
       }
       break;
+
+    case NANOARROW_TYPE_RUN_END_ENCODED: {
+      struct ArrowArrayView* run_ends_view;
+      run_ends_view = array_view->children[0];
+      if (run_ends_view->null_count != 0) {
+        ArrowErrorSet(error, "Run Ends cannot be null but found null_count value %ld",
+                      (long)run_ends_view->null_count);
+        return EINVAL;
+      }
+      for (int64_t i = 0, prev_run_end = 0; i < run_ends_view->length; i++) {
+        int64_t run_end = ArrowArrayViewGetIntUnsafe(run_ends_view, i);
+        if (run_end < 0 || run_end < prev_run_end) {
+          return EINVAL;
+        }
+        prev_run_end = run_end;
+        if (i == run_ends_view->length - 1 && run_end != array_view->length) {
```

Comment:
This "violation" is not really a violation. The parent array can change it's length without mutating the immutable run ends children.

Also keep in mind that if the parent has an offset, that last run-end would be affected and not match the logical length of the parent.